app/javascript/components/Greeting.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+
+const Greeting = () => {
+  const { greeting, isLoading, error } = useSelector((state) => state.greeting);
+  
+  if (error) {
+    return (
+      <>
+        <h2>Something went wront</h2>
+        <p>{ error }</p>
+      </>
+    )  
+  }
+  
+  isLoading ? 'Loading....' : (
+    <>
+       <div>{ greeting }</div>
+    </>
+  );
+}
+
+export default Greeting;
```

Comment:
- [ ] Great job so far. However, I got blank page while fired up your app. You need to use the useSelector to get access to your reducer from Greeting component and use useEffect to load it here not inside your App.js. Kindly move them to here and debug where necessary to make the browser displays the random greetings. Your Greeting component should look like this :point_down: 

```
import React, { useEffect } from 'react'
import { useSelector } from 'react-redux'
import { useDispatch } from 'react-redux';
import getRandomGreeting from '../redux/greeting/greetingSlice'

const Greeting = () => {
  const dispatch = useDispatch()
  const { greeting, loading } = useSelector((state) => state.greeting);
  console.log(greeting);
  
  useEffect(() => {
    dispatch(getRandomGreeting());
  }, [dispatch])
  
   if (loading) {
    return <h2>Loading...</h2>;
  }

  return (
    <>
    <div>
      <h1>Greetings</h1>
      <h2>{greeting.message}</h2>
    </div>
    </>
  )
}

export default Greeting;

```

![output](https://github.com/jecode93/hello-rails-react/assets/26597432/73eb655e-7113-4ed4-b88f-77124606dea9)